### PR TITLE
849 verify text case insensitive

### LINKED ISSUE #849: Verify string regardless of case
I have a two websites that I want to verify with the same code. Problem is one website the disclaim is in ALL CAPS and one is lowercase. Example

```
Use of this site constitutes acceptance of our

USE OF THIS SITE CONSTITUTES ACCEPTANCE OF OUR
```
So i propose the following api change:

```
Element Should Contain    locator, expected, message= 
```
to 

```
Element Should Contain    locator, expected, case_sensitive=True, message=
``` 

Default of course being True to avoid breaking changes and when the developer wants case insensitive, they can pass `case_sensitive=False`

I am willing to make these changes and submit a PR if it might be approved.

### PR DESCRIPTION
**Pull request by rubygeek**
Added code and tests to resolve  #849

### CODE REVIEW

File: src/SeleniumLibrary/keywords/element.py [hunk 1]
```diff
@@ -56,35 +56,54 @@ def element_should_contain(self, locator, expected, message=None):
...
+        The ``ignore_case`` argument can be set to True to compare case
+        insensitive, default is False.
+
         Use `Element Text Should Be` if you want to match the exact text,
         not a substring.
         """
-        actual = self.find_element(locator).text
+        actual = actual_before = self.find_element(locator).text
+        expected_before = expected
+
```
Comment (aaltat) [inline src/SeleniumLibrary/keywords/element.py]: We are not in favor of code blocks. Please remove the empty line from this line and from lines 71 and 76.

File: src/SeleniumLibrary/keywords/element.py [hunk 2]
```diff
@@ -56,35 +56,54 @@ def element_should_contain(self, locator, expected, message=None):
         The ``message`` argument can be used to override the default error
         message.
 
+        The ``ignore_case`` argument can be set to True to compare case
+        insensitive, default is False.
+
         Use `Element Text Should Be` if you want to match the exact text,
         not a substring.
```
Comment (aaltat) [inline src/SeleniumLibrary/keywords/element.py]: In new line, add text: ``ignore_case`` argument is new in SeleniumLibrary 3.1.

Please fix in keywords where the new argument is added.

File: src/SeleniumLibrary/keywords/element.py [hunk 3]
```diff
@@ -56,35 +56,54 @@ def element_should_contain(self, locator, expected, message=None):
...
 
         The ``message`` argument can be used to override the default error
         message.
+
+        The ``ignore_case`` argument can be set to True to compare case
+        insensitive, default is False.
         """
         actual = self.find_element(locator).text
+        expected_before = expected
+
```
Comment (aaltat) [inline src/SeleniumLibrary/keywords/element.py]: We are not in favor of code blocks. Please remove the empty line from this line and from line 99.

File: test/acceptance/keywords/content_assertions.robot [hunk 4]
```diff
@@ -138,10 +143,12 @@ Element Should Not Contain
 
 Element Text Should Be
     Element Text Should Be    some_id    This text is inside an identified element
+    Element Text Should Be    some_id    This TEXT IS INSIDE AN IDENTIFIED ELEMENT  ignore_case=True
     Run Keyword And Expect Error
     ...    The text of element 'some_id' should have been 'inside' but it was 'This text is inside an identified element'.
     ...    Element Text Should Be    some_id    inside
 
+
```
Comment (aaltat) [inline test/acceptance/keywords/content_assertions.robot]: Empty line here is not needed.

File: test/acceptance/keywords/content_assertions.robot [hunk 5]
```diff
@@ -119,15 +119,20 @@ Page Should Not Contain Element With Disabling Source Logging
 
 Element Should Contain
     Element Should Contain    some_id    This text is inside an identified element
```
Comment (aaltat) [inline test/acceptance/keywords/content_assertions.robot:121]: Please add line: `| Element Should Contain | some_id | This text is inside an identified element | ignore_case=False |`

File: test/acceptance/keywords/content_assertions.robot [hunk 6]
```diff
@@ -119,15 +119,20 @@ Page Should Not Contain Element With Disabling Source Logging
...
     Run Keyword And Expect Error
     ...    Element with locator 'missing_id' not found.
     ...    Element Should Contain    missing_id    This should report missing element.
+    Run Keyword And Expect Error
+    ...    Element 'some_id' should have contained text 'foobar' but its text was 'This text is inside an identified element'.
+    ...    Element Should Contain    some_id    foobar    ignore_case=True
 
 Element Should Not Contain
     Element Should Not Contain    some_id    This text is not inside an identified element
+    Element Should Not Contain    some_id    THIS TEXT is not inside an identified element   ignore_case=False
```
Comment (aaltat) [inline test/acceptance/keywords/content_assertions.robot]: 1) Please add line: `Element Should Not Contain | some_id | This text is not inside an identified element | ignore_case=True |`
2) Please also add new test where `Element Should Not Contain` when `ignore_case`argument is used with True and False values but the keyword fails. In the test use `Run Keyword And Expect Error`to verify that correct error is raised.

**Review by aaltat — CHANGES_REQUESTED** [review]
Code generally looks good, expect the code block which we do not usually use. But there should be few more test done before the PR can be merged.

**Comment by rubygeek** [comment]
I have fixed the code according to suggestions and ready for final review. @aaltat

**Review by aaltat — APPROVED** [review]

**Pull request merged**